ci-operator/step-registry/upi/vsphere/windows/pre/upi-vsphere-windows-pre-commands.sh

```diff
@@ -17,6 +17,18 @@ function createWindowsInstanceFile() {
       echo "$(date -u --rfc-3339=seconds) - created ${windows_instance_file}"
 }
 
+# Creates a folder at the given path if it does not exist.
+# Will return error if the parent folder also does not exist.
+# Takes the folder's path as an argument.
+function ensureSubfolderExists() {
+      folder=$1;
+      if ! govc folder.info $folder; then
+          echo "$(date -u --rfc-3339=seconds) - creating folder $folder..."
+          govc folder.create $folder
+	  echo "${folder}" >> "${SHARED_DIR}/created_folders.txt"
```

Comment:
`${SHARED_DIR}` is shared across all steps and the file can be overwritten, resulting in a resource leak. I suggest naming this file `windows_vm_folders.txt` to lower the collision rate with other files.

Reply:
+1 for above comment